### PR TITLE
feat(text-style): to fix Nested marks with styling are stripped by default #5720

### PR DESCRIPTION
**Pull request by mieze018**
## Changes Overview
Added functionality to merge styles of nested spans into child spans during HTML parsing.

## Implementation Approach
Introduced a new option `mergeNestedSpanStyles` in the `TextStyle` extension.

## Testing Done
Added demos for React and Vue. Added test cases for both examples.

## Verification Steps
Verify that test cases are sufficient.

## Checklist
- [x] I have created a [changeset](https://github.com/changesets/changesets) for this PR if necessary.
- [x] My changes do not break the library.
- [x] I have added tests where applicable.
- [x] I have followed the project guidelines.
- [x] I have fixed any lint issues.

## Related Issues
https://github.com/ueberdosis/tiptap/issues/5720

### CODE REVIEW

File: demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import '@tiptap/extension-text-style'
+
+import { Extension, Mark } from '@tiptap/core'
+
+export const AddGlobalAttributes = Extension.create({
```
Comment (bdbch) [inline demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts:5]: Are those extensions required for the demo? I thought this should also work by the default Color or TextStyle extension.

File: demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+              }
+            },
+          },
+        },
+      },
+    ]
+  },
+})
+
+export const Strong = Mark.create({
```
Comment (bdbch) [inline demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts:63]: Do we need this for the demo? Can't we use the default Bold extension from `@tiptap/extension-bold`?

**Review by bdbch — CHANGES_REQUESTED** [review]
I'm questioning the demo extensions here - the implementation itself looks fine.

I'd expect the demos not to require it's own extensions to test mark nested mark styles?

File: demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts (hunk 1 above)
Comment (mieze018) [reply demos/src/Marks/TextStyle/React/AddGlobalAttributes.ts]: Thank you for the review. I have rewritten the demo to use the default extensions. [aaab0b2](https://github.com/ueberdosis/tiptap/pull/5897/commits/aaab0b28dd378939fdc0ebaa25e812c65ee7607f)
I mistakenly assumed it should be tested with the TextStyle extension alone.

File: .changeset/tasty-sheep-compare.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@tiptap/extension-text-style": minor
+---
+
+Fixed nested marks with styling are stripped by default #5720
```
Comment (nperez0111) [inline .changeset/tasty-sheep-compare.md:5]: Should maybe mention the new option. This goes in the release notes

Comment (nperez0111) [reply .changeset/tasty-sheep-compare.md]: Looks great!

**Review by nperez0111 — APPROVED** [review]

**Pull request merged**